File: justlearning/palindrom.cpp

#include <iostream>

int main() {
    std::string word;
    std::cout << "Enter a word: " << std::endl;
    std::cin>> word;
    std::string fword= word;
    std::reverse(word.begin(), word.end());
    if (fword==word){
        std::cout<< fword << " is a palindrom.";
    }else{
        std::cout<< fword << " is not a palindrom.";
    }
    return 0;
}

File: rust/palindrom.rs
use std::io;

fn main(){
    println!("Enter a word: ");
    let mut word= String::new();
    io::stdin()
        .read_line(&mut word)
        .expect("");
    let fw= word.trim().chars().collect::<String>();
    let bw= word.trim().chars().rev().collect::<String>();
    if fw==bw{
        println!("{fw} is a palindrom");
    }else{
        println!("{fw} is not a palindrom");
    }
}
